Sporadic test failure in AuthorizationCommandInterceptorTest: unable to capture logs in time

From CI log:


```
Step #5 - "dse-6.8": INFO  [main] 2021-02-08 18:35:59,119 ExternalStorage.java:129 - About to run [engine:junit-jupiter]/[class:io.stargate.it.cql.AuthorizationCommandInterceptorTest]/[method:grantSelect()] with storage cluster DSE 6.8
Step #5 - "dse-6.8": INFO  [Exec Stream Pumper] 2021-02-08 18:35:59,124 ProcessRunner.java:76 - stargate9-0> WARN  [mainIOThread-0] 2021-02-08 18:35:59,121 LoggingAuthorizationProcessorImpl.java:46 - testing: addPermissions: cassandra, AL
LOW, ACCESS, [SELECT], AuthorizedResource{kind=TABLE, keyspace=auth_keyspace2, element=table3}, auth_user1
Step #5 - "dse-6.8": WARN  [main] 2021-02-08 18:35:59,132 ExternalStorage.java:138 - Error in [engine:junit-jupiter]/[class:io.stargate.it.cql.AuthorizationCommandInterceptorTest]/[method:grantSelect()]
Step #5 - "dse-6.8": java.lang.AssertionError: 
Step #5 - "dse-6.8": Expecting:
Step #5 - "dse-6.8":   <[]>
Step #5 - "dse-6.8": to contain exactly (and in same order):
Step #5 - "dse-6.8":   <["cassandra, ALLOW, ACCESS, [SELECT], AuthorizedResource{kind=TABLE, keyspace=auth_keyspace2, element=table3}, auth_user1"]>
Step #5 - "dse-6.8": but could not find the following elements:
Step #5 - "dse-6.8":   <["cassandra, ALLOW, ACCESS, [SELECT], AuthorizedResource{kind=TABLE, keyspace=auth_keyspace2, element=table3}, auth_user1"]>
Step #5 - "dse-6.8": 
Step #5 - "dse-6.8":    at io.stargate.it.cql.AuthorizationCommandInterceptorTest.grantSelect(AuthorizationCommandInterceptorTest.java:86)
Step #5 - "dse-6.8":    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
Step #5 - "dse-6.8":    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
Step #5 - "dse-6.8":    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Step #5 - "dse-6.8":    at java.lang.reflect.Method.invoke(Method.java:498)
Step #5 - "dse-6.8":    at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:688)
```

--- a/testing/src/test/java/io/stargate/it/cql/AuthorizationCommandInterceptorTest.java
+++ b/testing/src/test/java/io/stargate/it/cql/AuthorizationCommandInterceptorTest.java
@@ -72,12 +72,12 @@ public class AuthorizationCommandInterceptorTest extends BaseOsgiIntegrationTest
 
   private List<String> addedMsgs(String cql) {
     session.execute(cql);
-    return log.filter(0, Pattern.compile(".+testing: addPermissions: (.+)"), 1);
+    return log.filter(0, Pattern.compile(".+testing: addPermissions: (.+)"), 1, 1);
   }
 
   private List<String> removedMsgs(String cql) {
     session.execute(cql);
-    return log.filter(0, Pattern.compile(".+testing: removePermissions: (.+)"), 1);
+    return log.filter(0, Pattern.compile(".+testing: removePermissions: (.+)"), 1, 1);
   }
 
   @Test

--- a/testing/src/test/java/io/stargate/it/storage/LogCollector.java
+++ b/testing/src/test/java/io/stargate/it/storage/LogCollector.java
@@ -16,6 +16,7 @@
 package io.stargate.it.storage;
 
 import io.stargate.it.exec.OutputListener;
+import java.time.Duration;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
@@ -25,6 +26,7 @@ import java.util.concurrent.ConcurrentLinkedQueue;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
+import org.awaitility.Awaitility;
 import org.junit.jupiter.api.extension.ExtensionContext.Store;
 
 public class LogCollector implements OutputListener, Store.CloseableResource {
@@ -69,5 +71,12 @@ public class LogCollector implements OutputListener, Store.CloseableResource {
             })
         .filter(Objects::nonNull)
         .collect(Collectors.toList());
+  }
+
+  public List<String> filter(int node, Pattern pattern, int group, int numExpectedMessages) {
+    return Awaitility.await()
+        .atMost(Duration.ofMinutes(2))
+        .pollInterval(Duration.ofMillis(100))
+        .until(() -> filter(node, pattern, group), list -> list.size() >= numExpectedMessages);
   }
 }

